packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -35,6 +39,10 @@ class _DefaultHeroTag {
 /// If the [onPressed] callback is null, then the button will be disabled and
 /// will not react to touch.
 ///
+/// If the floating action bar is a descendant of a [Scaffold] that also has a
+/// [BottomAppBar], the [BottomAppBar] will show a notch to accomodate the
+/// [FloatingActionButton] when it overlaps the [BottomAppBar].
```

Comment:
done. I did not mention that it can be changed with `Scaffold.setFloatingActionButtonNotch` as I think the users of `FloatingActionButton` shouldn't care about how it tells the scaffold about the notch...